tools/tertiary-analysis/scanpy/scanpy_macros2.xml

```diff
@@ -155,18 +162,57 @@ EMBL-EBI https://www.ebi.ac.uk/ and Teichmann Lab at Wellcome Sanger Institute.
+        </param>
+      </when>
+      <when value=""/>
+    </conditional>
   </xml>
 
   <xml name="export_mtx_outputs">
-    <data name="matrix_10x" format="txt" from_work_dir="matrix.mtx" label="${tool.name} on ${on_string}: 10x matrix">
-      <filter>export_mtx</filter>
-    </data>
-    <data name="genes_10x" format="tsv" from_work_dir="genes.tsv" label="${tool.name} on ${on_string}: 10x genes">
-      <filter>export_mtx</filter>
-    </data>
-    <data name="barcodes_10x" format="tsv" from_work_dir="barcodes.tsv" label="${tool.name} on ${on_string}: 10x barcodes">
-      <filter>export_mtx</filter>
-    </data>
+    <collection name="mtx_raw" type="list">
+      <filter>export_mtx_inputs[export_mtx] == True and 
+        export_mtx_inputs[mtx_compression] == ""</filter>
+          <data name="matrix_10x" label="${tool.name} on ${on_string}: 10x matrix" format="txt" from_work_dir="matrix.mtx"/>
+          <data name="genes_10x" label="${tool.name} on ${on_string}: 10x genes" format="tsv" from_work_dir="genes.tsv"/>
+          <data name="barcodes_10x" label="${tool.name} on ${on_string}: 10x barcodes" format="tsv" from_work_dir="barcodes.tsv"/>
+    </collection>
+    <collection name="mtx_zip" type="list" format="zip">
+      <filter>export_mtx_inputs[export_mtx] == True and 
+        export_mtx_inputs[mtx_compression] == "zip"</filter>
+          <data name="matrix_10x_zip" label="${tool.name} on ${on_string}: 10x matrix (zip)" from_work_dir="matrix.mtx.zip"/>
+          <data name="genes_10x_zip" label="${tool.name} on ${on_string}: 10x genes (zip)" from_work_dir="genes.tsv.zip"/>
+          <data name="barcodes_10x_zip" label="${tool.name} on ${on_string}: 10x barcodes (zip)" from_work_dir="barcodes.tsv.zip"/>
+    </collection>
+    <collection name="mtx_gzip" type="list" format="gz">
+      <filter>export_mtx_inputs[export_mtx] == True and 
+        export_mtx_inputs[mtx_compression] == "gzip"</filter>
+          <data name="matrix_10x_gzip" label="${tool.name} on ${on_string}: 10x matrix (gzip)" from_work_dir="matrix.mtx.gz"/>
+          <data name="genes_10x_gzip" label="${tool.name} on ${on_string}: 10x genes (gzip)" from_work_dir="genes.tsv.gz"/>
+          <data name="barcodes_10x_gzip" label="${tool.name} on ${on_string}: 10x barcodes (gzip)" from_work_dir="barcodes.tsv.gz"/>
+    </collection>
+    <collection name="mtx_bz2" type="list" format="bz2">
+      <filter>export_mtx_inputs[export_mtx] == True and 
+        export_mtx_inputs[mtx_compression] == "bz2"</filter>
+          <data name="matrix_10x_bz2" label="${tool.name} on ${on_string}: 10x matrix (bz2)" from_work_dir="matrix.mtx.bz2"/>
+          <data name="genes_10x_bz2" label="${tool.name} on ${on_string}: 10x genes (bz2)" from_work_dir="genes.tsv.bz2"/>
+          <data name="barcodes_10x_bz2" label="${tool.name} on ${on_string}: 10x barcodes (bz2)" from_work_dir="barcodes.tsv.bz2"/>
+    </collection>
+    <collection name="mtx_zstd" type="list" format="zst">
```

Comment:
I cannot find zst support in Galaxy, I would suggest that we comment this part. If we really need it, we could add it as a format to the galaxy codebase (it is called a datatype in Galaxy-speak).